karyon3-rxnetty/src/main/java/com/netflix/karyon/rxnetty/HttpRoutingRequestHandler.java

```diff
@@ -0,0 +1,54 @@
+package com.netflix.karyon.rxnetty;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.reactivex.netty.protocol.http.server.HttpServerRequest;
+import io.reactivex.netty.protocol.http.server.HttpServerResponse;
+import io.reactivex.netty.protocol.http.server.RequestHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import rx.Observable;
+
+/**
+ * Very basic URL routing using a set of HttpEndpointDefinition to specify the routes and 
+ * associated handler.
+ * 
+ * @author elandau
+ *
+ */
+public class HttpRoutingRequestHandler implements RequestHandler<ByteBuf, ByteBuf> {
+    private final static Logger LOG = LoggerFactory.getLogger(HttpRoutingRequestHandler.class);
+    
+    private final List<HttpEndpointDefinition> defs = new ArrayList<>();
+    
+    public HttpRoutingRequestHandler(Set<HttpEndpointDefinition> endpoints) throws Exception {
+        defs.addAll(endpoints);
```

Comment:
It doesn't seem like this will work for having a generic catch all with overrides for more specific paths. For example:

```
/foo/* -> fooHandler
/bar/* -> barHandler
/*     -> genericHandler
```

Since these get added to a `Set` there isn't a well defined order and in the handle method the first getting added to the list wins. So in the example above anything added after `/*` will get ignored and there is no way to control which order they will get added in.